### PR TITLE
bgp: T3322: Fix timers for neighbor

### PR DESCRIPTION
**Pull request by sever-sever**
## Change Summary

Fix timers for neighbor.
We lost it in the template.
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

T3322
## Component(s) name

bgp
## Proposed changes

Add timers for bgp neighbors, It was lost after rewriting bgp to python format.
## How to test
VyOS config:
```
set protocols bgp 64501 neighbor 203.0.113.1 remote-as '65001'
set protocols bgp 64501 neighbor 203.0.113.1 timers connect '234'
set protocols bgp 64501 neighbor 203.0.113.1 timers holdtime '62'
set protocols bgp 64501 neighbor 203.0.113.1 timers keepalive '20'
```
Should generate FRR config:
```
!
router bgp 64501
 neighbor 203.0.113.1 remote-as 65001
 neighbor 203.0.113.1 timers 20 62
 neighbor 203.0.113.1 timers connect 234
!

```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Review by c-po — CHANGES_REQUESTED** [review]
Why is the config.timers is not none required atvthe end?

**Comment by sever-sever** [comment]
> Why is the config.timers is not none required atvthe end?

Not sure. I delete  "config.timers is not none"

**Comment by c-po** [comment]
Can we have connect timers without other timers?

File: data/templates/frr/bgp.frr.tmpl [hunk 1]
```diff
@@ -58,6 +58,10 @@
 {%   if config.ttl_security is defined and config.ttl_security.hops is defined and config.ttl_security.hops is not none %}
  neighbor {{ neighbor }} ttl-security hops {{ config.ttl_security.hops }}
 {%   endif %}
+{%   if config.timers is defined and config.timers.connect is defined and config.timers.holdtime is defined and config.timers.keepalive is defined %}
```
Comment (c-po) [inline data/templates/frr/bgp.frr.tmpl]: With this `{% if ... %}` block one can only have a BGP connect timer when also a holdtime and keepalive is defined. FRR supports setting the BGP connect time independently from the holdtime and keepalive.

Please make BGP connect time configurable independently from holdtime and keepalive.

**Review by c-po — CHANGES_REQUESTED** [review]

File: data/templates/frr/bgp.frr.tmpl (hunk 1 above)
Comment (sever-sever) [reply data/templates/frr/bgp.frr.tmpl]: @c-po done

Comment (c-po) [reply data/templates/frr/bgp.frr.tmpl]: This is a bit too complicated/too many nesting levels. Please shorten this to:

```
{%   if config.timers is defined %}
{%     if config.timers.connect is defined and config.timers.connect is not none %}
 neighbor {{ neighbor }} timers connect {{ config.timers.connect }}
{%     endif %}
{%     if config.timers.holdtime is defined and config.timers.keepalive is defined and config.timers.holdtime is not none and config.timers.keepalive is not none %}
 neighbor {{ neighbor }} timers {{ config.timers.keepalive }} {{ config.timers.holdtime }}
{%     endif %}
{%   endif %}
```

**Review by c-po — CHANGES_REQUESTED** [review]

File: data/templates/frr/bgp.frr.tmpl (hunk 1 above)
Comment (sever-sever) [reply data/templates/frr/bgp.frr.tmpl]: Done
